src/ui/component/tag/tag.tsx

```diff
@@ -1,12 +1,35 @@
-import type { FC } from 'react'
+import { useCallback, type FC } from 'react'
+import { Icon } from '@/ui/component/icon/icon'
 import './tag.scss'
+import { useTranslation } from 'react-i18next'
 
 type TagProps = {
   tagName: string
+  onDelete?: () => void
 }
 
-export const Tag: FC<TagProps> = ({ tagName }) => (
-  <div className="okp4-dataverse-portal-tag-main">
-    <div className="okp4-dataverse-portal-tag-name">{tagName}</div>
-  </div>
-)
+export const Tag: FC<TagProps> = ({ tagName, onDelete }) => {
+  const { t } = useTranslation('common')
+  const handleDelete = useCallback(() => {
+    if (onDelete) {
+      onDelete()
+    }
+  }, [onDelete])
+
+  return (
+    <div className="okp4-dataverse-portal-tag-main">
+      <div className="okp4-dataverse-portal-tag-name">{tagName}</div>
+      {onDelete && (
+        <div
+          className="okp4-dataverse-portal-tag-icon"
+          onClick={handleDelete}
```

Comment:
You've already made an assertion on the `onDelete` function, so you can give it directly to the event handler.
```suggestion
          onClick={onDelete}
```